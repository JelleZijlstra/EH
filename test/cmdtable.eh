#!/usr/bin/ehi
# Test defining commands by manipulating the cmdtable directly.

commands->'test' = printvar

$test -abde -cfgh=42 --test --foo='bar' --"foox"="barx" --"bam"=[3, 4]

$test 'foo' 'bar'
