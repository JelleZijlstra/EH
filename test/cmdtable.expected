%%stdout%%
@array [
	'_ehphp' => @bool true
	'a' => @bool true
	'b' => @bool true
	'bam' => @array [
		0 => @int 3
		1 => @int 4
	]
	'c' => @int 42
	'd' => @bool true
	'e' => @bool true
	'f' => @int 42
	'foo' => @string "bar"
	'foox' => @string "barx"
	'g' => @int 42
	'h' => @int 42
	'test' => @bool true
]
@array [
	'_ehphp' => @bool true
	0 => @string "foo"
	1 => @string "bar"
]
%%stderr%%
